Assets/MixedRealityToolkit.SDK/Experimental/Features/Support/Scripts/BaseServiceManager.cs

```diff
@@ -0,0 +1,395 @@
+            }
+
+            return registered;
+        }
+
+        /// <inheritdoc />
+        public bool RegisterService<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityService
+        {
+            T serviceInstance = ActivateInstance<T>(concreteType, supportedPlatforms, args);
+            if (serviceInstance == null)
+            {
+                return false;
+            }
+
+            return RegisterService<T>(serviceInstance);
+        }
+
+        /// <inheritdoc />
+        public bool UnregisterDataProvider<T>(string name = null) where T : IMixedRealityDataProvider
+        {
+            T dataProviderInstance = GetDataProvider<T>(name);
+            return UnregisterDataProvider<T>(dataProviderInstance);
+        }
+
+        /// <inheritdoc />
+        public bool UnregisterDataProvider<T>(T dataProviderInstance) where T : IMixedRealityDataProvider
+        {
+            if ((dataProviderInstance == null) || (!dataProviders.Contains(dataProviderInstance))) { return false; }
+
+            dataProviders.Remove(dataProviderInstance);
+            return true;
+        }
+
+        /// <inheritdoc />
+        public bool UnregisterService<T>(string name = null) where T : IMixedRealityService
+        {
+            T serviceInstance = FindService<T>(name);
+
+            if (serviceInstance == null) { return false; }
+
+            return UnregisterService<T>(serviceInstance);
+        }
+
+        /// <inheritdoc />
+        public bool UnregisterService<T>(T serviceInstance) where T : IMixedRealityService
+        {
+            if (serviceInstance == null) { return false; }
+
+            Type interfaceType = typeof(T);
+            if (!registeredServices.ContainsKey(interfaceType)) { return false; }
+
+            registeredServices.Remove(interfaceType);
+            return MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
+        }
+
+        /// <summary>
+        /// Activates an instance of the specified concrete type using the provided argument collection.
+        /// </summary>
+        /// <typeparam name="T">The interface which must be implemented by the concrete type.</typeparam>
+        /// <param name="concreteType">The type of object to be instantiated.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the concrete type is supported.</param>
+        /// <param name="args">Collection of arguments to provide to the concrete type's constructor.</param>
+        /// <returns>An instance of the concrete type. Returns a default value of T (typically null) in the event of a failure.</returns>
+        private T ActivateInstance<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityService
+        {
+            if (concreteType == null) { return default(T); }
```

Comment:
>default(T) [](start = 47, length = 10)

Does this have to be default(T) because the compiler thinks we might be using structs or something else here? Otherwise it would be great to just be explicit and be like, we return null in this case.

Reply:
> Does this have to be default(T) because the compiler thinks we might be using structs or something else here?

Exactly